src/fs-utils.js

```diff
@@ -0,0 +1,109 @@
+import { copy } from 'fs-extra';
+import { Promise } from 'bluebird';
+import temp from 'temp';
+import fs from 'fs';
+
+const log = require('debug')('electron-windows-installer:fs-utils');
+
+temp.track();
+
+const promisedCopy = Promise.promisify(copy);
+const promisedCreateTempDir = Promise.promisify(temp.mkdir);
+
+export { promisedCopy as copy, promisedCreateTempDir as createTempDir };
+
+export function renameAsync(oldPath, newPath) {
+  return new Promise((resolve, reject) => {
+    fs.rename(oldPath, newPath, err => {
+      if (err) {
+        log(err);
+        reject(err);
+      }
+
+      resolve();
+    });
+  });
+}
+
+export function inspectAsync(path) {
+  return new Promise((resolve, reject) => {
+    fs.stat(path, (err, stats) => {
+      if (err) {
+        log(err);
+        reject(err);
+        return;
+      }
+
+      resolve(stats);
+    });
+  });
+}
+
+export async function existsFileAsync (file) {
+  let stats;
+
+  try {
+    stats = await inspectAsync(file);
+    return stats.isFile();
+  } catch(err) {
+    log(err);
+  }
+
+  return false;
+}
+
+export function writeFileAsync(file, data) {
+  return new Promise((resolve, reject) => {
+    fs.writeFile(file, data, err => {
+      if (err) {
+        log(err);
+        reject(err);
+        return;
+      }
+
+      resolve();
+    });
+  });
+}
+
+export function readFileAsync(file) {
```

Comment:
Makes sense. Promisify looks much better for native `fs` functions. I'll change it with upcoming patch.